mithril-client/src/snapshot_downloader.rs

```diff
@@ -143,15 +188,23 @@ impl SnapshotDownloader for HttpSnapshotDownloader {
     async fn probe(&self, location: &str) -> MithrilResult<()> {
         debug!(self.logger, "HEAD Snapshot location='{location}'.");
 
-        let request_builder = self.http_client.head(location);
-        let response = request_builder.send().await.with_context(|| {
-            format!("Cannot perform a HEAD for snapshot at location='{location}'")
-        })?;
+        if let Some(local_path) = Self::file_scheme_to_local_path(location) {
+            if fs::metadata(local_path).is_ok() {
+                Ok(())
+            } else {
+                Err(anyhow!("Local snapshot location='{location}' not found"))
+            }
```

Comment:
This could be a bit simplified, and as a bonus it keep the error of the `fs::metadata` in the returned result, adding more context in case of a failure.
```suggestion
            fs::metadata(local_path)
                .with_context(|| format!("Local snapshot location='{location}' not found"))?;
            Ok(())
```

Reply:
Fixed ([diff](https://github.com/input-output-hk/mithril/compare/650f2db55039b38a69fb62a928862369d6ba0d17..775b7d68874108dd146724eff0a7f0179509837d)).